### ISSUE TITLE
 HdmiCEC

### ISSUE THREAD

**Issue by Marchella1983** [zgemma-star/meta-zgemma#53]
Hi, please fix HdmiCEC.

On H9 Combo and Twin when using HdmiCEC, the TV does not turn on from the TV remote control, on H7 it all works...

**Comment by zgemma-star** [comment]
TV and box are in deepsleep?
The TV remote is used to activate the TV. 
The TV will send CEC message to wakeup the box.

The problem is that the TV is activate and will direct fallback in deepsleep again?

**Comment by Marchella1983** [comment]
to H9 settings like H7...

Except it doesn't work on H9, I tried different settings for H9, but the TV doesn't turn on from my LG 55UJ630V 4K UHD remote control...

I repeat on H7 it works without any problems...
![1_0_1_697A_1B_FE_2300000_0_0_0_20191103195022](https://user-images.githubusercontent.com/28537448/68086978-3a0aef00-fe73-11e9-8479-782287bea8bc.jpg)

**Comment by zgemma-star** [comment]
"TV doesn't turn on from my LG 55UJ630V 4K UHD remote control."

So you use the LG remote to turn on your LG TV and this doesn't work?
So what your saying is that probably a CEC command blocks to startup the TV?

**Comment by Marchella1983** [comment]
> So what your saying is that probably a CEC command blocks to startup the TV?

Yeah, I'm turning on the LG remote and it's not working...

**Comment by artseluiko** [comment]
> So what your saying is that probably a CEC command blocks to startup the TV?

No, not the TV. TV can be turned on by LG Remote. The problem is that H9 does not wake up from standby via hdmi cec commands

**Comment by andreew75** [comment]
Hello developers!
This problem is still relevant.
I have also strange behavior with HDMI-CEC on my Zgemma H9 Combo. The receiver is connected to the TV Samsung M series 2018.
I would like the Zgemma box to go to standby when the TV goes into standby and wake up when the TV wakes up. But the following happens: Zgemma box goes into standby when the TV goes into standby, but it does not wake up from sleep when the TV wakes up.
Moreover, it is possible to bring the receiver out of standby mode from the receiver's remote control only the second time, that is, when the Power button is pressed for a split second, the signal lights up for a split second and immediately red, and when the Power button is pressed again, the receiver starts up. But if you disable HDMI CEC in the settings, then the inclusion occurs the first time. This issue occurs on all images, on any HDMI CEC setting. 
I also found that this problem occurs with many owners of Zgemma. More details here https://www.techkings.org/threads/zgemma-h9-2s-and-hdmi-cec-with-samsung-tv.154948/.
Could you fix this bug and fix it in the next driver release?
thanks

**Comment by ccristal** [comment]
> 
> 
> Hello developers!
> This problem is still relevant.
> I have also strange behavior with HDMI-CEC on my Zgemma H9 Combo. The receiver is connected to the TV Samsung M series 2018.
> I would like the Zgemma box to go to standby when the TV goes into standby and wake up when the TV wakes up. But the following happens: Zgemma box goes into standby when the TV goes into standby, but it does not wake up from sleep when the TV wakes up.
> Moreover, it is possible to bring the receiver out of standby mode from the receiver's remote control only the second time, that is, when the Power button is pressed for a split second, the signal lights up for a split second and immediately red, and when the Power button is pressed again, the receiver starts up. But if you disable HDMI CEC in the settings, then the inclusion occurs the first time. This issue occurs on all images, on any HDMI CEC setting.
> I also found that this problem occurs with many owners of Zgemma. More details here https://www.techkings.org/threads/zgemma-h9-2s-and-hdmi-cec-with-samsung-tv.154948/.
> Could you fix this bug and fix it in the next driver release?
> thanks

The very same happens on Zgemma H9.2S. I reported it nearly one year ago on [this post](https://www.techkings.org/threads/zgemma-h9-2s-and-hdmi-cec-with-samsung-tv.154948/), which contains screenshots and logs.

**Comment by zgemma-star** [comment]
The box does execute the CEC codes.
It seems some TV's do wake up with the commands Enigma2 sends and others don't.
Please make a log of CEC command to see what your TV transmit and Enigma2 transmit.
It seems enigma2 always sends a code "36" which means all hardware must go into standby and if it's in standby then wakeup.
When this doesn't work on your TV you must ask Samsung for a fix.

First activate the log and check if code 36 is send and otherwise place the log here.

**Comment by artseluiko** [comment]
In standby mode the box does not execute the code. It executes the code only if wakeup the box by zgemma remote.
It's not a Samsung problem, because this also happens with LG and other TVs.

**Comment by ccristal** [comment]
> The box does execute the CEC codes.
> It seems some TV's do wake up with the commands Enigma2 sends and others don't.
> Please make a log of CEC command to see what your TV transmit and Enigma2 transmit.
> It seems enigma2 always sends a code "36" which means all hardware must go into standby and if it's in standby then wakeup.
> When this doesn't work on your TV you must ask Samsung for a fix.
> 
> First activate the log and check if code 36 is send and otherwise place the log here.

You've got it backwards... :-)
We're not talking about the TV not being able to wake up when the box is woken up. It's the exact opposite: the Zgemma box does not wake up via HDMI-CEC when the TV wakes up, using the TV remote, whatever value you set "Handle wake-up from TV" to.
Therefore, it's not about what the Zgemma _sends_, it's about what it _receives_ (or doesn't).
The logs are [here](https://www.techkings.org/threads/zgemma-h9-2s-and-hdmi-cec-with-samsung-tv.154948/).

**Comment by artseluiko** [comment]
@ccristal I know what I am talking about. I mean that Zgemma box in standby mode can't handle CEC command. I said nothing about  
> it's not about what the Zgemma sends, it's about what it receives (or doesn't).
Please read patiently or use google translate if you don't understand what I am saying

**Comment by ccristal** [comment]
@artseluiko I didn't quote your post, did I? My reply was to @zgemma-star, not to you. Learn to read.

**Comment by andreew75** [comment]
> It's not a Samsung problem, because this also happens with LG and other TVs.

It's true! Because my Formuler F 1 works fine with Samsung TV, but when I connect the Zgemma box to it, this Zgemma box does not wake up. The problem is clearly at Zgemma!

**Comment by zgemma-star** [comment]
https://www.cec-o-matic.com/

Post both logs from Formuler and Zgemma.

Zgemma sends the commands what is given by Enigma2. Deepsleep CEC wakeup isn't support yet. Standby wakeup all the commado's are send. Problem if not works should been search in Enigma2 code.

**Comment by Tassel8062** [comment]
I still see this behaviour on zgemma h9 twin with Sony bravia tv

**Comment by andreew75** [comment]
I understand that the manufacturer will not fix this problem, and he is not interested in the buyer's problems. It's very sad

**Comment by deep666** [comment]
i have the same issue i just bought zgemma h9 combo and the box is working fine with TVs remote control everythink is OK but the wake up signal for the box is not OK. TV is going to standby also the stb but the stb is not waking up only TV.. with the zgemma remote all is working fine only bug is with the TVs remote control.... my tv is LG ... before i had octagon sf4008 with it everything was working fine so it must be a bug in zgemmmas firmware , software..... i tried all the images out there and nothing ...

**Comment by elnoelsi** [comment]
H9S needs to press power on twice of decode remote to wake up+turn on tv

pressing it once doesnt turn on tv

**Comment by Marcus051333** [comment]
Why is this bug still present?
